[backport] [dashing] Prevent rviz_rendering::AssimpLoader from loading materials twice (#622)

hidmic opened the pull request:
Backport #622.

Dashing CI (up to `rviz2`, `rviz_rendering`, and `rviz_rendering_tests`):

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=13214)](http://ci.ros2.org/job/ci_linux/13214/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=8143)](http://ci.ros2.org/job/ci_linux-aarch64/8143/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=10928)](http://ci.ros2.org/job/ci_osx/10928/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=13229)](http://ci.ros2.org/job/ci_windows/13229/)

hidmic committed:
Prevent rviz_rendering::AssimpLoader from loading materials twice. (#622)

Retrieve material if it already exists. Assumes resource path uniqueness.

Signed-off-by: Michel Hidalgo <michel@ekumenlabs.com>

hidmic commented:
`Dpr` job failed with:

```
Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

I'll retry later.

nuclearsandwich commented:
@hidmic it looks like there were build errors in the CI jobs as well. I'll work on the docker rate limit issue if you can take a look at those.

hidmic commented:
Looks like Dashing API is slightly different. Have to tweak this patch.

hidmic committed:
Use Ogre::static_pointer_cast instead of std::static_pointer_cast

In Ogre v1.10.12, Ogre::SharedPtr is not a std::shared_ptr subclass
but a separate implementation of it.

Signed-off-by: Michel Hidalgo <michel@ekumenlabs.com>

hidmic commented:
FYI @nuclearsandwich.

hidmic commented:
Warnings on MacOS and Windows appear unrelated. Ready to merge when you are @nuclearsandwich.

jacobperron reviewed (approved):
LGTM
